### PR TITLE
W-14145955: Modularization of extensions support

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: modules/extensions-support/pom.xml [hunk 1]
```diff
@@ -17,6 +17,30 @@
     <properties>
         <javaModuleName>org.mule.runtime.extensions.support</javaModuleName>
         <!-- Required by mule-extension-ast-loader -->
+
+        <surefire.args>
+            -XX:+IgnoreUnrecognizedVMOptions
+            --add-modules=net.bytebuddy.agent
+            --add-opens=org.mule.runtime.extensions.support/org.mule.runtime.module.extension.internal.loader.validation=org.mule.runtime.core
```
Comment (mbuchwald) [inline modules/extensions-support/pom.xml:24]: (This ones is a test package)

File: modules/extensions-support/pom.xml [hunk 2]
```diff
@@ -17,6 +17,31 @@
     <properties>
         <javaModuleName>org.mule.runtime.extensions.support</javaModuleName>
         <!-- Required by mule-extension-ast-loader -->
+
+        <surefire.args>
+            -XX:+IgnoreUnrecognizedVMOptions
+            --add-modules=net.bytebuddy.agent
+            --add-reads=org.mule.runtime.extensions.support=com.google.gson
```
Comment (mbuchwald) [inline modules/extensions-support/pom.xml:24]: Only used in test

File: modules/extensions-support/src/main/java/module-info.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+/**
+ * Default implementation of the Mule Extension API.
+ *
+ * @moduleGraph
+ * @since 4.6
+ */
+module org.mule.runtime.extensions.support {
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/module-info.java:13]: add the annotation with the privileged packages

File: modules/extensions-support/src/main/java/module-info.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  opens org.mule.runtime.module.extension.internal.runtime.connectivity.oauth.authcode to
+      org.mule.runtime.core;
+  opens org.mule.runtime.module.extension.internal.runtime.execution to
+      org.mule.runtime.core;
+  opens org.mule.runtime.module.extension.internal.runtime.operation to
+      org.mule.runtime.core;
+  opens org.mule.runtime.module.extension.internal.runtime.resolver to
+      org.mule.runtime.core;
+  opens org.mule.runtime.module.extension.internal.runtime.source to
+      org.mule.runtime.core;
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/module-info.java:99]: why these opens?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**